test/wit.js

```diff
@@ -0,0 +1,158 @@
+      });
+      const component = await componentNew(generatedComponent);
+
+      // Transpile the component
+      const { files, imports, exports } = await transpile(component);
+      deepStrictEqual(imports, [
+        "test:feature-gates/foo",
+        ]);
+      deepStrictEqual(exports, [
+        ["foo", "instance"],
+        ["test:feature-gates/foo@0.2.1", "instance"],
+      ]);
+      ok(files['component.js'], "component js was generated");
+      ok(files['component.d.ts'], "component typings were generated");
+      ok(files['interfaces/test-feature-gates-foo.d.ts'], "interface typings were generated");
+
+      // Check the interfaces file for the right exports
+      const interfaces = Buffer.from(files['interfaces/test-feature-gates-foo.d.ts']).toString('utf8');
+      ok(interfaces.includes("export function a(): void;"), "unconstrained export foo/a is present");
+      ok(interfaces.includes("export function b(): void;"), "@since(0.2.1) export foo/b is present (version matches)");
+      ok(interfaces.includes("export function c(): void;"), "@since(0.2.1) export foo/c is present (feature is ignored)");
+      ok(interfaces.includes("export function d(): void;"), "@unstable(...) export is present, with all features enabled");
+    });
+
+    // feature-gates: unstable should be present with only the specific feature enabled
+    test("Feature gates (single feature enabled)", async () => {
+      // Build a dummy WIT component
+      const generatedComponent = await componentEmbed({
+        witSource: featureGatesWitContent,
+        dummy: true,
+        metadata: [
+          ["language", [["javascript", ""]]],
+          ["processed-by", [["dummy-gen", "test"]]],
+        ],
+        features: {
+          tag: "list",
+          val: [
+            "fancier-foo",
+          ],
+        },
+      });
+      const component = await componentNew(generatedComponent);
+
+      // Transpile the component
+      const { files, imports, exports } = await transpile(component);
+      deepStrictEqual(imports, [
+        "test:feature-gates/foo",
+        ]);
+      deepStrictEqual(exports, [
+        ["foo", "instance"],
+        ["test:feature-gates/foo@0.2.1", "instance"],
+      ]);
+      ok(files['component.js'], "component js was generated");
+      ok(files['component.d.ts'], "component typings were generated");
+      ok(files['interfaces/test-feature-gates-foo.d.ts'], "interface typings were generated");
+
+      // Check the interfaces file for the right exports
+      const interfaces = Buffer.from(files['interfaces/test-feature-gates-foo.d.ts']).toString('utf8');
+      ok(interfaces.includes("export function a(): void;"), "unconstrained export foo/a is present");
+      ok(interfaces.includes("export function b(): void;"), "@since(0.2.1) export foo/b is present (version matches)");
+      ok(interfaces.includes("export function c(): void;"), "@since(0.2.1) export foo/c is present (feature is ignored)");
```

Comment:
Again this one seems like it shouldn't be included?

Reply:
Yeah `c()` was `feature = gated`, right? But due to `@since(version = 0.2.1)` it's "stabilized" which means it should be visible... 

AFAIK this is working as intended, with the main level for visibility being the `@unstable` -> `@since` transition

I'll bring this up during the CM meeting tomorrow